packages/toolpad-components/src/Text.tsx

```diff
@@ -23,9 +22,22 @@ interface TextProps extends Omit<BaseProps, 'children'> {
 
 const MarkdownContainer = styled('div')({
   display: 'block',
-  [`&:empty::before`]: { content: '""', display: 'inline-block' },
+  maxWidth: 'fill-available',
+  '& span:empty::before': {
+    content: '""',
+    display: 'inline-block',
+  },
 });
 
+const CodeContainer = styled('pre')(({ theme }) => ({
+  backgroundColor: theme.palette.grey[200],
+  marginLeft: theme.spacing(1),
+  marginRight: theme.spacing(1),
+  borderRadius: theme.shape.borderRadius,
+  padding: theme.spacing(1),
+  overflowX: 'scroll',
+}));
```

Comment:
In the future, when we look into performance, we could look if styling `& pre` in the container isn't faster.